### PR TITLE
Budget Refactor

### PR DESCRIPTION
**Pull request by lalexgap**
Updates the `budget` based on the discussion in stand up and https://www.notion.so/State-Channels-4cc770686d05438bb23b3c37f4069d7e?p=ce435f483e044d8291a8a01f570391d4

### CODE REVIEW

File: packages/client-api-schema/src/types.ts [hunk 1]
```diff
@@ -167,28 +162,43 @@ export type ChallengeChannelRequest = JsonRpcRequest<'ChallengeChannel', {channe
...
+
 export interface TokenBudget {
   token: string;
-  pending: Balance;
-  free: Balance;
-  inUse: Balance;
-  direct: Balance;
+  availableReceiveCapacity: string;
+  availableSendCapacity: string;
+  channels: ChannelBudget[];
```
Comment (lalexgap) [inline packages/client-api-schema/src/types.ts]: I switched this from a `Record` as specified in the notion doc to keep it consistent with `budgets`. I think we should either using a `Record` or `Array` consistently in the API types but I'm ok using either depending on what's easiest for `web3-torrent`.

File: packages/client-api-schema/src/types.ts [hunk 2]
```diff
@@ -167,28 +162,43 @@ export type ChallengeChannelRequest = JsonRpcRequest<'ChallengeChannel', {channe
 export type ChallengeChannelResponse = JsonRpcResponse<ChannelResult>;
 
 // Budget
+
+export interface ChannelBudget {
+  channelId: string;
+  status: 'pending' | 'available';
```
Comment (tomclose) [inline packages/client-api-schema/src/types.ts]: I think we decided that we didn't need the `status` - you can check the channel status to find this out.

File: packages/channel-client/tests/fakes/fake-channel-provider.ts [hunk 3]
```diff
@@ -312,20 +312,18 @@ export class FakeChannelProvider implements ChannelProviderInterface {
...
-    const {hub, site, playerAmount, hubAmount} = params;
-
+  private async approveBudgetAndFund(params: TokenBudgetRequest): Promise<SiteBudget> {
     // TODO: Does this need to be delayed?
-    const result = {
-      hub: hub.signingAddress,
-      site,
+    const result: SiteBudget = {
+      hubAddress: params.hub.signingAddress,
+      domain: 'web3torrent.statechannels.org',
```
Comment (geoknee) [inline packages/channel-client/tests/fakes/fake-channel-provider.ts]: Should we have a reference to web3torrent baked into the fake channel provider? Other apps such as RPS might (do) use it.

File: packages/client-api-schema/src/types.ts (hunk 2 above)
Comment (geoknee) [reply packages/client-api-schema/src/types.ts]: 2f84c95

File: packages/channel-client/tests/fakes/fake-channel-provider.ts (hunk 3 above)
Comment (geoknee) [reply packages/channel-client/tests/fakes/fake-channel-provider.ts]: ecd818d

**Pull request merged**